packages/pluggableWidgets/screenshottaker-native/src/ScreenshotTaker.tsx

```diff
@@ -0,0 +1,115 @@
+            };
+        };
+    };
+
+const timeout = (ms: number) => new Promise(resolve => setTimeout(resolve, ms));
+
+export function ScreenshotTaker(props: ScreenshotTakerProps<ScreenshotTakerType>): ReactElement | null {
+    const captureContent = props.whatToCapture === "captureContent";
+    const viewRef = useRef(null);
+    const shouldTakeScreenshotRef = useRef(false);
+    const [shouldTakeScreenshot, setShouldTakeScreenshot] = useState(false);
+    const styles = flattenStyles(defaultScreenshotTakerStyles, props.style);
+    const g = global as CustomGlobal;
+
+    // Runs on Unmount
+    useEffect(() => {
+        return () => {
+            if (g.screenshotRunner?.task) {
+                (global as CustomGlobal).screenshotRunner.task.isRunning = false;
+            }
+        };
+    }, []);
+
+    // Runs on dependency change
+    useEffect(() => {
+        if (
+            (global as CustomGlobal).screenshotRunner?.task?.isRunning &&
+            !shouldTakeScreenshotRef.current &&
+            !shouldTakeScreenshot
+        ) {
+            shouldTakeScreenshotRef.current = true;
+            setShouldTakeScreenshot(true);
+        }
+    }, [shouldTakeScreenshotRef.current, setShouldTakeScreenshot]);
+
+    useEffect(() => {
+        if (
+            shouldTakeScreenshot &&
+            props.pageName.status === ValueStatus.Available &&
+            props.base64.status === ValueStatus.Available &&
+            props.onScreenshot?.canExecute
+        ) {
+            setShouldTakeScreenshot(false);
+            (async () => {
+                const options: CaptureOptions = {
+                    result: "base64",
+                    quality: 1
+                };
+
+                if (props.beforeScreenshot?.canExecute) {
+                    executeAction(props.beforeScreenshot);
+                    await timeout(3000); // Wait for action to complete
+                }
+
+                await timeout(1000); // Wait for data to load
+                if (captureContent) {
+                    captureRef(viewRef, options).then(onCapture).catch(onCaptureFailure);
+                } else {
+                    captureScreen(options).then(onCapture).catch(onCaptureFailure);
+                }
+            })();
+        }
+    }, [shouldTakeScreenshot, props.beforeScreenshot, props.pageName, props.base64, props.onScreenshot]);
+
+    const onCapture = useCallback(
+        async (base64: string) => {
+            if (base64.length <= 0) {
+                console.error(`Screenshot [${g.screenshotRunner.task.name}] failed. Base64 string is empty.`);
+            }
+            const formattedBase64 = base64.replace(/\s/g, "").trim();
+            props.pageName.setValue(g.screenshotRunner.task.name);
+            props.base64.setValue(formattedBase64);
+            executeAction(props.onScreenshot);
+        },
+        [props.pageName, props.base64, props.onScreenshot]
+    );
+
+    function onCaptureFailure(e: Error) {
```

Comment:
Wrap this one in a useCallback as  this is being used in the useEffect

Reply:
done